demos/python_demos/object_detection_demo/object_detection_demo.py

```diff
@@ -0,0 +1,297 @@
+            xmin = max(int(detection.xmin), 0)
+            ymin = max(int(detection.ymin), 0)
+            xmax = min(int(detection.xmax), size[1])
+            ymax = min(int(detection.ymax), size[0])
+            class_id = int(detection.id)
+            det_label = labels[class_id] if labels and len(labels) >= class_id else '#{}'.format(class_id)
+            log.info('{:^9} | {:10f} | {:4} | {:4} | {:4} | {:4} '
+                     .format(det_label, detection.score, xmin, ymin, xmax, ymax))
+
+
+def main():
+    args = build_argparser().parse_args()
+
+    log.info('Initializing Inference Engine...')
+    ie = IECore()
+
+    plugin_config = get_plugin_configs(args.device, args.num_streams, args.num_threads)
+
+    log.info('Loading network...')
+
+    model = get_model(ie, args)
+    has_landmarks = args.architecture_type == 'retina'
+
+    detector_pipeline = AsyncPipeline(ie, model, plugin_config, logger=log,
+                                      device=args.device, max_num_requests=args.num_infer_requests)
+
+    try:
+        input_stream = int(args.input)
+    except ValueError:
+        input_stream = args.input
+    cap = cv2.VideoCapture(input_stream)
+    if not cap.isOpened():
+        log.error('OpenCV: Failed to open capture: ' + str(input_stream))
+        sys.exit(1)
+
+    next_frame_id = 0
+    next_frame_id_to_show = 0
+
+    log.info('Starting inference...')
+    print("To close the application, press 'CTRL+C' here or switch to the output window and press ESC key")
+
+    palette = ColorPalette(len(model.labels) if model.labels else 100)
+    presenter = monitors.Presenter(args.utilization_monitors, 55,
+                                   (round(cap.get(cv2.CAP_PROP_FRAME_WIDTH) / 4),
+                                    round(cap.get(cv2.CAP_PROP_FRAME_HEIGHT) / 8)))
+
+    metrics = PerformanceMetrics()
+
+    while cap.isOpened():
+        # 1. Process all completed requests
+        results = detector_pipeline.get_result(next_frame_id_to_show)
+        if results:
+            objects, frame_meta = results
+            frame = frame_meta['frame']
+            start_time = frame_meta['start_time']
+
+            if len(objects) and args.raw_output_message:
+                print_raw_results(frame.shape[:2], objects, model.labels, args.prob_threshold)
+
+            presenter.drawGraphs(frame)
+            frame = draw_detections(frame, objects, palette, model.labels, args.prob_threshold, has_landmarks)
+            metrics.update(start_time, frame)
+            if not args.no_show:
+                cv2.imshow('Detection Results', frame)
+                key = cv2.waitKey(1)
+
+                ESC_KEY = 27
+                # Quit.
+                if key in {ord('q'), ord('Q'), ESC_KEY}:
+                    break
+                presenter.handleKey(key)
+            next_frame_id_to_show += 1
+            continue
+
+        # 2. Wait for empty request
+        detector_pipeline.await_any()
```

Comment:
This should be more sophisticated. If `detector_pipeline.await_any()` blocked, there are new results became available. You need proceed to the next loop iteration before reading next frame to check whether that new result has `next_frame_id_to_show` id.

Reply:
Rearranged, please check

Reply:
@Wovchena ping